ExchangeRateApp/Controllers/ExchangeRateViewController.swift

```diff
@@ -7,93 +7,108 @@
+        }
+        
+        components?.queryItems = queryItems
+        
+        guard let url = components?.url else {
+            return
+        }
+        
+        print("🛠 Created URL", url)
         
-        let task = URLSession.shared.dataTask(with: url) { (data, response, error) in
+        let task = URLSession.shared.dataTask(with: url) { data, response, error in
             if error != nil {
                 print("ERROR")
             } else {
-                if let content = data{
+                if let data = data {
                     do {
-                        let myJson = try JSONSerialization.jsonObject(with: content, options: JSONSerialization.ReadingOptions.mutableContainers) as AnyObject
-                        print(myJson)
-                        
-                        let currency = "\(myJson)"
-                        let startIndex = currency.index(currency.startIndex, offsetBy: 16)
-                        let endIndex = currency.index(currency.startIndex, offsetBy: 21)
-                        let currencyString = String(currency[startIndex...endIndex])
-                        self.exchange.append(Code(code: "", fullName: "", currency: currencyString))
-                        print(currencyString)
-                        } catch {
+                        let jsonObject = try JSONSerialization.jsonObject(with: data) as? [String: Double]
+                    
+                        if let exchangeRates: [ExchangeRate] = jsonObject?.map({ currencyPair, exchangeRate in
+                            let lhsCurrency = String(currencyPair.prefix(3))
+                            let rhsCurrency = String(currencyPair.suffix(3))
+                            
+                            print("🛠 1 \(lhsCurrency) exchangeable for \(exchangeRate) \(rhsCurrency)")
+                            
+                            return ExchangeRate(fromCurrency: lhsCurrency, toCurrency: rhsCurrency, rate: exchangeRate)
+                        }) {
+                            self.exchangeRates = exchangeRates
+                            
+                            DispatchQueue.main.async {
+                                self.tableView.reloadData()
+                            }
+                        }
+                    } catch {
                         
                     }
                 }
             }
         }
-        task.resume()
-        
         
+        task.resume()
     }
     
-
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+        if let currencyPairViewController = segue.destination as? CurrencyPairViewController {
+            currencyPairViewController.delegate = self
```

Comment:
Set `delegate` property, otherwise when it is called `delegate?.pickedCurrency(item)` it is `nil`